core/arch/arm/plat-stm32mp1/conf.mk

```diff
@@ -5,17 +5,25 @@ flavor_dts_file-157C_DHCOM_PDK2 = stm32mp157c-dhcom-pdk2.dts
 flavor_dts_file-157C_DK2 = stm32mp157c-dk2.dts
 flavor_dts_file-157C_ED1 = stm32mp157c-ed1.dts
 flavor_dts_file-157C_EV1 = stm32mp157c-ev1.dts
+flavor_dts_file-157A_DK1_SCMI = stm32mp157a-dk1-scmi.dts
+flavor_dts_file-157C_DK2_SCMI = stm32mp157c-dk2-scmi.dts
+flavor_dts_file-157C_ED1_SCMI = stm32mp157c-ed1-scmi.dts
+flavor_dts_file-157F_EV1_SCMI = stm32mp157c-ev1-scmi.dts
```

Comment:
The change could be more local if you used an additional variable (i.e., no change to the platform flavors). For example `ifeq (y,$(CFG_STM32MP1_RCC_SECURE))` ... append the `-scmi` part to the file name ... `endif`. The test could even use some existing `CFG_SCMI_*` variable perhaps?

Reply:
The downside of this would be to specify `CFG_STM32MP1_RCC_SECURE=y` in the command line but I guess that's ok.

Reply:
We expect the DT sets the platform configuration. So I would prefer have specific DTSes for specific configuration, instead of a config switch that would override the DT content.

Reply:
I guess Jérome's comment still stand as we can keep the flavor of the non-scmi DT and add the `-scmi` for the `CFG_EMBED_DTB_SOURCE_FILE`. Anyway, not a strong opinion.

Reply:
I could remove the `157?_*_SCMI` flavors but I still need to enumerate in some flavorlist to get `CFG_STM32MP15` and other switch set (incl. DDR size). In the end, it is simpler to define `PLATFORM_FLAVOR` values.

Reply:
> I could remove the `157?_*_SCMI` flavors but I still need to enumerate in some flavorlist to get `CFG_STM32MP15` and other switch set (incl. DDR size). In the end, it is simpler to define `PLATFORM_FLAVOR` values.

I don't understand this, because you are only adding the new flavors to lists where the non-SCMI flavor is already present. Can you give an example?

Reply:
I rather have mp15xx-xxx.dts files in opte_os related to mp15xx-xxx.dts in Linux/U-Boot,
and mp15xx-xxx-scmi.dts in optee_os for the bards mp15xx-xxx-scmi.dts defined in Linux/U-Boot.

Being listed in a flavour list prefers one from providing fewother needed config switches:
https://github.com/OP-TEE/optee_os/blob/4.0.0/core/arch/arm/plat-stm32mp1/conf.mk#L53-L74
https://github.com/OP-TEE/optee_os/blob/4.0.0/core/arch/arm/plat-stm32mp1/conf.mk#L153-L156

Reply:
> I rather have mp15xx-xxx.dts files in opte_os related to mp15xx-xxx.dts in Linux/U-Boot,
and mp15xx-xxx-scmi.dts in optee_os for the bards mp15xx-xxx-scmi.dts defined in Linux/U-Boot.

 I am fine with having separate DTS files for the "RCC is secure" case.

> Being listed in a flavour list prefers one from providing fewother needed config switches: https://github.com/OP-TEE/optee_os/blob/4.0.0/core/arch/arm/plat-stm32mp1/conf.mk#L53-L74 https://github.com/OP-TEE/optee_os/blob/4.0.0/core/arch/arm/plat-stm32mp1/conf.mk#L153-L156

Ah, here's the problem I think. IMO `$(flavorlist-xxx)` should be a flavor list (!), not a list of DTS files. See for example: https://github.com/OP-TEE/optee_os/blob/4.0.0/core/arch/arm/plat-imx/conf.mk#L1-L235.
Remember that the platform flavor is visible as a C macro (`PLATFORM_FLAVOR_<flavorname>`) so changing it can  impact on the code.

This is just about maintenance, so if you think your proposal is more convenient, I'm OK with that.